test/functional/apps/discover/_discover.js

```diff
@@ -126,6 +126,19 @@ export default function({ getService, getPageObjects }) {
         const isVisible = await PageObjects.discover.hasNoResults();
         expect(isVisible).to.be(false);
       });
+
+      it('should reload the saved search with persisted query to show the initial hit count', async function() {
+        // apply query some changes
+        await queryBar.setQuery('test');
+        await queryBar.submitQuery();
+        // reset to persisted state
```

Comment:
Are there any hits for the 'test' query?  I can't tell from this if it's 0, 14,004, or somewhere in between.  I think the test would be better if it verified some count between 0 and 14,004 before resetting the search.

Reply:
Good point, thx, I've added this additional verification 👍

Reply:
Sorry but I just realized that reset search doesn't clear the search but resets it to the query that may have been saved in a saved search.  Should we change the test to start with a saved search which has a query, change the query, verify the count, and reset search back to the saved query?  My concern is that it could just clear the query, which would be wrong.
But I guess you know that `$route.reload();` will do the right thing.  I don't know the code so I'm going to review and leave the test decision up to you.

Reply:
Makes sense, so I've refactored it to function that way, started a flaky test suite runner to check: https://kibana-ci.elastic.co/job/kibana+flaky-test-suite-runner/467/